### PR TITLE
Improvements Needed for Quickstart

### PR DESCRIPTION
**Pull request by mdemoret-nv**
There are some changes that are necessary to test the Quickstart guide from the conda package. This PR makes those changes so that once they are merged, a conda package can be built allowing for the QSG to be tested.

Some improvements:
- Improves the python build system in CMake to be fully generic
   - This follows the design built and used in Morpheus
   - Improves on the design by allowing multiple python packages to be built in the same CMake project
   - Also adds python stub files to SRF

### CODE REVIEW

**Review by dagardner-nv — APPROVED** [review]
LGTM

File: include/srf/node/rx_sink.hpp [hunk 1]
```diff
@@ -49,7 +50,10 @@ template <typename T, typename ContextT>
 class RxSink : public RxSinkBase<T>, public RxRunnable<ContextT>, public RxPrologueTap<T>
 {
   public:
-    using observer_t = rxcpp::observer<T>;
+    using observer_t       = rxcpp::observer<T>;
+    using on_next_fn_t     = std::function<void(T)>;
+    using on_error_fn_t    = std::function<void(std::exception_ptr)>;
+    using on_complete_fn_t = std::function<void()>;
```
Comment (ryanolson) [inline include/srf/node/rx_sink.hpp:56]: these don't appear to be used immediately `RxSink`, would they be better in `RxSinkBase`?

**Review by drobison00 — APPROVED** [review]
LGTM -- Everything looked reasonable in the quick start; if we're just building two new packages should be fine.

File: include/srf/node/rx_sink.hpp (hunk 1 above)
Comment (ryanolson) [reply include/srf/node/rx_sink.hpp:56]: I guess `RxSink` is probably the right choice, since the `RxSource` observer is used in `RxNode`

**Review by ryanolson — APPROVED** [review]

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**